ruby/engine/embedded_help.rb

```diff
@@ -637,45 +639,61 @@ def self.glob(pattern, *args, **options)
       pattern_array = pattern
     end
 
-    #puts "Dir.glob pattern = #{pattern}, pattern_array = #{pattern_array}, args = #{args}, options = #{options}"
+    pattern_has_embedded = pattern_array.any? {|p| p.to_s.chars.first == ':'}
+    base_has_embedded = (!base.nil? && base.to_s.chars.first == ':')
+    if !pattern_has_embedded && !base_has_embedded
+      # puts "Original glob"
+      return self.original_glob(pattern, flags: flags, base: base, sort: sort)
+    end
+
+    absolute_base = if base.nil?
+                      nil
+                    else
+                      OpenStudio.get_absolute_path(base)
+                    end
+    if debug
+      puts "pattern_array=#{pattern_array}"
+      puts "base=#{base}"
+      puts "flags=#{flags}"
+      puts "pattern_has_embedded=#{pattern_has_embedded}"
+      puts "base_has_embedded=#{base_has_embedded}"
+      puts "absolute_base=#{absolute_base}"
+    end
+
+    # DLM: seems like this is needed for embedded paths, possibly due to leading ':' character?
     override_args_extglob = false
 
     result = []
     pattern_array.each do |pattern|
 
-      if pattern.to_s.chars.first == ':'
-
-        # DLM: seems like this is needed for embedded paths, possibly due to leading ':' character?
-        override_args_extglob = true
-
-        #puts "searching embedded files for #{pattern}"
-        absolute_pattern = OpenStudio.get_absolute_path(pattern)
-        #puts "absolute_pattern #{absolute_pattern}"
-
-        EmbeddedScripting::fileNames.each do |name|
-          absolute_path = OpenStudio.get_absolute_path(name)
+      absolute_pattern = if pattern.to_s.chars.first == ':'
+                           OpenStudio.get_absolute_path(pattern)
+                         else
+                           pattern
+                         end
```

Comment:
Now, the pattern needs to be expanded only if it's an embedded one.

Eg if this is '*.gemspec', don't expand it to `$(pwd)/*.gemspec` which is going to be on your actual host filesystem always.

(Note: I suppose doing File.expand_path(pattern, base) was a possibility)